src/res_printer.ml

```diff
@@ -553,7 +553,7 @@ let printConstant ?(templateLiteral = false) c =
   | Pconst_float (s, _) -> Doc.text s
   | Pconst_char c ->
     let str =
-      match c with
+      match Char.unsafe_chr c with
```

Comment:
There's still a `unsafe_char` usage, but I think this one is safe because it's just value testing and no I/O is involved.

Reply:
It seems okay, but there may be some duplication with string_of_int_as_char, you may do a clean up later after this commit landed